src/components/SelectInput/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+    _cs,
+    listToMap,
+} from '@togglecorp/fujs';
+
+import SelectInputContainer from '../SelectInputContainer';
+import { rankedSearchOnList } from '../../utils';
+
+import styles from './styles.css';
+
+type OptionKey = string | number;
+interface Option {
+    [index: string]: any;
+}
+
+export interface SelectInputProps<T extends OptionKey, K> {
+    value: T,
+    onChange: (newValue: T) => void;
+    options: Option[],
```

Comment:
`Options` should be a generic variable IMO

Reply:
Option is a generic Object

Reply:
No it is not.
Generic is equivalent to templates in C++